File: java/src/main/java/de/thorbenkuck/kafka/KafkaConsumer.java
package de.thorbenkuck.kafka;

import de.thorbenkuck.kafka.data.ExampleMessage;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

@Component
public class KafkaConsumer {

    @KafkaListener(topics = "${kafka.input.topic}", containerFactory = "exampleMessageContainerFactory")
    public void listen(ExampleMessage exampleMessage) {
        System.out.println("We received: " + exampleMessage.getMessage());
    }

}

File: kotlin/src/main/kotlin/de/thorbenkuck/kafka/KafkaConsumer.kt
package de.thorbenkuck.kafka

import de.thorbenkuck.kafka.data.ExampleMessage
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Component

@Component
class KafkaConsumer {

	@KafkaListener(
			topics = ["\${kafka.input.topic}"],
			containerFactory = "exampleMessageContainerFactory"
	)
	fun listen(exampleMessage: ExampleMessage) {
		println("We received: " + exampleMessage.message)
	}
}